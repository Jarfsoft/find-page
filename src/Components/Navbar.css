.nav {
  display: flex;
  justify-content: space-around;
  text-align: center;
  font-size: small;
}

.nav a {
  text-decoration: none;
  color: gray;
  margin: 1.5rem 0;
}

.head {
  font-size: small;
  font-weight: bold;
  margin: 1.5rem 0;
}
