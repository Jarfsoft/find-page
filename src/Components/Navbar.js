import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';

import actions from '../Actions/index';
import './Navbar.css';

function Navbar() {
  const dispatch = useDispatch()
  const header = useSelector((state) => state.navbar.page);
  const user = useSelector((state) => state.navbar.user);
  const signout = () => {
    dispatch(actions.user(0));
  };
  const rightButton = () => {
    if(header === 'Devices')
      return (user !== 0 ? <a onClick={signout}>Sign out</a> : <Link to={'/start'}>Get in</Link>);
    else
      return (<Link to={'/'}>Home</Link>);
  }

  return (
    <nav>
      <Link to={'/favorites'}>Favorites</Link>
      <h1>{header ? header : ''}</h1>
      {rightButton()}
    </nav>
  );
}

export default Navbar;
