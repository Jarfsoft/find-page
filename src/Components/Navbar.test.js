import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { BrowserRouter } from 'react-router-dom';
import Navbar from './Navbar';

describe('Navbar render', () => {
  const mockStore = configureStore();
  let store;
  const state = {
    navbar: {
      page: 'Devices',
    },
    users: {
      userId: 0,
      name: '',
    },
  };

  describe('Snapshot check', () => {
    it('should render Navbar section', () => {
      store = mockStore(state);
      const navbar = renderer
        .create(
          <Provider store={store}>
            <BrowserRouter>
              <Navbar />
            </BrowserRouter>
          </Provider>,
        ).toJSON();
      expect(navbar).toMatchSnapshot();
    });
  });

  describe('Navbar contains', () => {
    it('should have a text that says Devices', () => {
      const navbar = render(
        <Provider store={store}>
          <BrowserRouter>
            <Navbar />
          </BrowserRouter>
        </Provider>,
      );
      const header = navbar.container.querySelector('h1');
      expect(header.textContent).toBe('Devices');
    });
    it('should contain 2 link', () => {
      const navbar = render(
        <Provider store={store}>
          <BrowserRouter>
            <Navbar />
          </BrowserRouter>
        </Provider>,
      );
      const link = navbar.container.querySelectorAll('a');
      expect(link).toHaveLength(2);
    });
  });
});
