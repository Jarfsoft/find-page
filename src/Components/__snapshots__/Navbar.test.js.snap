// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar render Snapshot check should render Navbar section 1`] = `
<nav
  className="nav"
>
  <a
    href="/favorites"
    onClick={[Function]}
  >
    Favorites
  </a>
  <h1
    className="head"
  >
    Devices
  </h1>
  <a
    href="/start"
    onClick={[Function]}
  >
    Get in
  </a>
</nav>
`;
